packages/fuel-gauge/src/abi/abi-coder.test.ts

```diff
@@ -0,0 +1,1236 @@
+import { Contract, FuelError, Interface } from 'fuels';
+import type {
+  AssetId,
+  BigNumberish,
+  DecodedValue,
+  EvmAddress,
+  RawSlice,
+  WalletUnlocked,
+} from 'fuels';
+import { expectToThrowFuelError, launchTestNode } from 'fuels/test-utils';
+
+import { AbiContract, AbiContractFactory } from '../../test/typegen';
+import { EnumWithNativeInput, ExternalEnumInput } from '../../test/typegen/contracts/AbiContract';
+import type {
+  EnumWithBuiltinTypeInput,
+  EnumWithBuiltinTypeOutput,
+  EnumWithVectorInput,
+  EnumWithVectorOutput,
+  IdentityInput,
+  IdentityOutput,
+  StructDoubleGenericInput,
+  StructSimpleInput,
+  StructSimpleOutput,
+  StructWithGenericArrayInput,
+  StructWithMultiOptionInput,
+  StructWithMultiOptionOutput,
+  StructCInput,
+  StructWithNestedArrayInput,
+  StructWithNestedTupleInput,
+  StructSingleGenericInput,
+} from '../../test/typegen/contracts/AbiContract';
+import type { Option, Result, Vec } from '../../test/typegen/contracts/common';
+
+import {
+  U16_MAX,
+  U16_MIN,
+  U256_MAX,
+  U256_MIN,
+  U32_MAX,
+  U32_MIN,
+  U64_MAX,
+  U64_MIN,
+  U8_MAX,
+  U8_MIN,
+} from './constants';
+import { toEqualBn } from './vitest.matcher';
+
+expect.extend({ toEqualBn });
+
+/**
+ * @group node
+ */
+describe('AbiCoder', () => {
+  let contract: AbiContract;
```

Comment:
I'd favor creating a `setupContract()` helper like [here](https://github.com/FuelLabs/fuels-ts/blob/8321098963996088ba33988176e028a2af5da98b/packages/fuel-gauge/src/coverage-contract.test.ts#L37) and using it in every test instead of reusing the same contract/node across all tests.
```ts
using contract = await setupContract();
```
There's nothing wrong with the approach you've taken here given that all these calls are read-only, but if we decide to parallelize these tests in the future we'll have to do the `setupContract` approach or we'd be getting issues related to already spent coins. It's better to just have a hard rule of always spinning up a node for every test.